activerecord/test/cases/excluding_test.rb

```diff
@@ -22,6 +22,16 @@ def test_result_set_does_not_include_collection_of_excluded_records
     assert_not_includes relation, posts(:thinking)
   end
 
+  def test_result_set_does_not_include_collection_of_excluded_records_from_a_query
+    query = Post.where(id: @post)
+
+    result_set = assert_sql(/(SELECT "posts"."id" FROM "posts")|(SELECT `posts`.`id` FROM `posts`)/) do
+      Post.excluding(query)
+    end
+
+    assert_not_includes result_set, @post
```

Comment:
```suggestion
    assert_sql(/SELECT #{ActiveRecord::Base.connection.quote_column_name('posts.id')} FROM/) do
      query = Post.where(id: @post)
      posts = Post.excluding(query)
      assert_not_includes posts, @post
    end
```

Reply:
With this you do not need to change `assert_sql` method and regexp suites for multiple database adapters at the same time.

Reply:
@fatkodima I tried the following diff:

```diff
diff --git a/activerecord/test/cases/excluding_test.rb b/activerecord/test/cases/excluding_test.rb
index 83b1fc2b22..1bd58ce939 100644
--- a/activerecord/test/cases/excluding_test.rb
+++ b/activerecord/test/cases/excluding_test.rb
@@ -25,7 +25,7 @@ def test_result_set_does_not_include_collection_of_excluded_records
   def test_result_set_does_not_include_collection_of_excluded_records_from_a_query
     query = Post.where(id: @post)
 
-    assert_sql(/(SELECT "posts"."id" FROM "posts")|(SELECT `posts`.`id` FROM `posts`)/) do
+    assert_sql(/SELECT #{ActiveRecord::Base.connection.quote_column_name("posts.id")} FROM/) do
       result_set = Post.excluding(query)
 
       assert_not_includes result_set, @post
@@ -51,7 +51,7 @@ def test_result_set_through_association_does_not_include_collection_of_excluded_
   def test_result_set_through_association_does_not_include_collection_of_excluded_records_from_a_relation
     relation = @post.comments
 
-    assert_sql(/(SELECT "comments"."id" FROM "comments")|(SELECT `comments`.`id` FROM `comments`)/) do
+    assert_sql(/SELECT #{ActiveRecord::Base.connection.quote_column_name("comments.id")} FROM/) do
       result_set = Comment.excluding(relation)
 
       assert_not_empty result_set
```

Unfortunately, the call to `quote_column_names` failed the tests:

```
Failure:
ExcludingTest#test_result_set_does_not_include_collection_of_excluded_records_from_a_query [/Users/seanpdoyle/Work/rails/rails/activerecord/test/cases/excluding_test.rb:28]:
Query pattern(s) /SELECT "posts.id" FROM/ not found.
Queries:
SELECT "posts"."id" FROM "posts" WHERE "posts"."id" = ?
SELECT 1 AS one FROM "posts" WHERE "posts"."id" != ? AND "posts"."id" = ? LIMIT ?.
Expected [/SELECT "posts.id" FROM/] to be empty?.


bin/test test/cases/excluding_test.rb:25

..F

Failure:
ExcludingTest#test_result_set_through_association_does_not_include_collection_of_excluded_records_from_a_relation [/Users/seanpdoyle/Work/rails/rails/activerecord/test/cases/excluding_test.rb:54]:
Query pattern(s) /SELECT "comments.id" FROM/ not found.
Queries:
SELECT "comments"."id" FROM "comments" WHERE "comments"."post_id" = ?
SELECT 1 AS one FROM "comments" WHERE "comments"."id" NOT IN (?, ?) LIMIT ?
SELECT "comments".* FROM "comments" WHERE "comments"."id" NOT IN (?, ?)
SELECT "comments".* FROM "comments" WHERE "comments"."post_id" = ?.
Expected [/SELECT "comments.id" FROM/] to be empty?.


bin/test test/cases/excluding_test.rb:51
```

Reply:
Sorry, I meant to use `quote_table_name` instead. There are a few examples of such uses in the activerecord tests already.

Reply:
Done!